### PR TITLE
Terraform (HCL) (*.tf): new parser

### PR DESCRIPTION
**Pull request by Haggus**
Add support for Terraform (HCL) files.

Continuation of https://github.com/universal-ctags/ctags/pull/2952

### CODE REVIEW

File: optlib/terraform.ctags [hunk 1]
```diff
@@ -25,11 +25,11 @@
 --kinddef-terraform=p,Provider,Terraform Provider
 --kinddef-terraform=m,Module,Terraform Module
 --kinddef-terraform=o,Output,Terraform Output
---regex-terraform=/^[[:space:]]*resource[[:space:]]*"([^"]*)"[[:space:]]*"([^"]*)"/\2/r,Resource/
---regex-terraform=/^[[:space:]]*data[[:space:]]*"([^"]*)"[[:space:]]*"([^"]*)"/\2/d,Data/
---regex-terraform=/^[[:space:]]*variable[[:space:]]*"([^"]*)"/\1/v,Variable/
---regex-terraform=/^[[:space:]]*provider[[:space:]]*"([^"]*)"/\1/p,Provider/
---regex-terraform=/^[[:space:]]*module[[:space:]]*"([^"]*)"/\1/m,Module/
---regex-terraform=/^[[:space:]]*output[[:space:]]*"([^"]*)"/\1/o,Output/
+--regex-terraform=/^resource[[:space:]]"([^"]+)"[[:space:]]"([^"]+)"/\2/r,Resource/
```
Comment (masatake) [inline optlib/terraform.ctags:28]: You use both kind letter and kind long name in "r,Resource".
However, `r` is already defined with `--kinddef-terraform`. So you should use only 'r' here.
```
--regex-terraform=/^resource[[:space:]]"([^"]+)"[[:space:]]"([^"]+)"/\2/r/
```

File: optlib/terraform.ctags [hunk 2]
```diff
@@ -16,13 +16,20 @@
...
---langdef=tf
---map-tf=+tf:.tf.tfvars
---regex-tf=/^[[:space:]]*resource[[:space:]]*"([^"]*)"[[:space:]]*"([^"]*)"/\2/r,Resource/
---regex-tf=/^[[:space:]]*data[[:space:]]*"([^"]*)"[[:space:]]*"([^"]*)"/\2/d,Data/
---regex-tf=/^[[:space:]]*variable[[:space:]]*"([^"]*)"/\1/v,Variable/
---regex-tf=/^[[:space:]]*provider[[:space:]]*"([^"]*)"/\1/p,Provider/
---regex-tf=/^[[:space:]]*module[[:space:]]*"([^"]*)"/\1/m,Module/
---regex-tf=/^[[:space:]]*output[[:space:]]*"([^"]*)"/\1/o,Output/
---regex-tf=/^([a-z0-9_]+)[[:space:]]*=/\1/f,TFVar/
+--langdef=terraform
```
[the remark this answers is not in the document]
Comment (Haggus) [reply optlib/terraform.ctags:19]: You are right. It should be Terraform.

**Comment by Haggus** [comment]
> Looks good for me. @Haggus thank you for adding the test cases. They were really what I needed.
> 
> I would like to squash these into one.
> 
> I would like to squash into one commit. In that case, we have to think about how the "author" of the commit should be.
> 
> The Author must be Antony Southworth. So your named will be put to the commit like:
> 
> ```
> Co-authored-by: Mateusz Mrowiec <...>
> Co-authored-by: Masatake YAMATO <...>
> ```
> 
> Is this acceptable?

Sounds good to me :+1:

**Comment by masatake** [comment]
Ok, I will take over this.

**Comment by Haggus** [comment]
@masatake One more thing. Windows build doesn't pass. I think entries in `ctags_vs2013.vcxproj` have been put incorrectly. I need to fix that.

**Comment by Haggus** [comment]
It's all green, good to go!

**Comment by masatake** [comment]
Closed via #3684.

**Pull request closed without merging**